lib/src/wiggle_abi/req_impl.rs

```diff
@@ -267,6 +267,30 @@ impl FastlyHttpReq for Session {
             "http"
         };
 
+        let ca_certs =
+            if (scheme == "https") && backend_info_mask.contains(BackendConfigOptions::CA_CERT) {
```

Comment:
As mentioned elsewhere to @acw , none of the other parameters check the scheme.
Do we want to add it to the other places, possibly in a separate PR?

Reply:
Since it's a bit out of scope for the PR, we can possibly do this as a cleanup pass later.

Reply:
👍 wfm.